src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -172,49 +212,156 @@ private BoundPattern BindPropertyPattern(PropertyPatternSyntax node, BoundExpres
             return new BoundPropertyPattern(node, type, boundPatterns, hasErrors: hasErrors);
         }
 
-        private ImmutableArray<BoundSubPropertyPattern> BindSubPropertyPatterns(PropertyPatternSyntax node, TypeSymbol type, DiagnosticBag diagnostics)
+        private ImmutableArray<BoundSubPropertyPattern> BindSubPropertyPatterns(
+            PropertyPatternSyntax node,
+            TypeSymbol type,
+            DiagnosticBag diagnostics)
         {
             var result = ArrayBuilder<BoundSubPropertyPattern>.GetInstance();
-            foreach (var syntax in node.PatternList.SubPatterns)
+            foreach (var e in node.SubPatterns)
             {
-                var propName = syntax.Left;
-                BoundPattern pattern;
-                HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-                LookupResultKind resultKind;
-                Symbol property = FindPropertyOrFieldByName(type, propName, out resultKind, ref useSiteDiagnostics);
-                if ((object)property != null)
+                var syntax = e as IsPatternExpressionSyntax;
+                var identifier = syntax?.Expression as IdentifierNameSyntax;
+                if (identifier == null) throw ExceptionUtilities.UnexpectedValue(syntax?.Expression.Kind() ?? e.Kind());
+                var propName = identifier.Identifier;
+                var boundMember = BindPropertyPatternMember(type, identifier, diagnostics);
+                var boundPattern = BindPattern(syntax.Pattern, null, boundMember.Type, boundMember.HasErrors, diagnostics);
+                result.Add(new BoundSubPropertyPattern(e, boundMember, boundPattern, boundPattern.HasErrors));
+            }
+
+            return result.ToImmutableAndFree();
+        }
+
+        // returns BadBoundExpression or BoundObjectInitializerMember
```

Comment:
Should the comment say BoundPropertyPatternMember instead of BoundObjectInitializerMember?